designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/NussknackerAppFactory.scala

```diff
@@ -1,119 +1,72 @@
-import pl.touk.nussknacker.ui.util.IOToFutureSttpBackendConverter
-import sttp.client3.SttpBackend
+import pl.touk.nussknacker.ui.metrics.RepositoryGauges
+import pl.touk.nussknacker.ui.process.repository._
+import pl.touk.nussknacker.ui.server.{AkkaHttpBasedRouteFactory, NussknackerHttpServer}
 import sttp.client3.asynchttpclient.cats.AsyncHttpClientCatsBackend
 
 import java.time.Clock
-
-object NussknackerAppFactory {
-
-  def apply(designerConfigLoader: DesignerConfigLoader): NussknackerAppFactory = {
-    new NussknackerAppFactory(designerConfigLoader)
-  }
-
-}
+import scala.concurrent.Future
 
 class NussknackerAppFactory(
     designerConfigLoader: DesignerConfigLoader,
 ) extends LazyLogging {
 
   def createApp(clock: Clock = Clock.systemUTC()): Resource[IO, Unit] = {
     for {
-      alreadyLoadedConfig           <- Resource.eval(designerConfigLoader.loadDesignerConfig())
-      managersDirs                  <- Resource.eval(IO.delay(alreadyLoadedConfig.managersDirs()))
-      deploymentManagersClassLoader <- DeploymentManagersClassLoader.create(managersDirs)
-      system                        <- createActorSystem(alreadyLoadedConfig.rawConfig)
-      executionContextWithIORuntime <- ExecutionContextWithIORuntimeAdapter.createFrom(system.dispatcher)
+      _ <- Resource.eval(IO(JavaClassVersionChecker.check()))
+      _ <- Resource.eval(IO(SLF4JBridgeHandlerRegistrar.register()))
+
+      alreadyLoadedConfig <- Resource.eval(designerConfigLoader.loadDesignerConfig())
+
+      actorSystem                   <- createActorSystem(alreadyLoadedConfig)
+      executionContextWithIORuntime <- ExecutionContextWithIORuntimeAdapter.createFrom(actorSystem.dispatcher)
       ioSttpBackend                 <- AsyncHttpClientCatsBackend.resource[IO]()
-      modelClassLoaderProvider = createModelClassLoaderProvider(
-        alreadyLoadedConfig.processingTypeConfigs.configByProcessingType,
-        deploymentManagersClassLoader
-      )
-      processingTypeConfigsLoader = createProcessingTypeConfigsLoader(
-        alreadyLoadedConfig,
-        ioSttpBackend
-      )(executionContextWithIORuntime.ioRuntime)
-      processingTypeDataLoader = new ProcessingTypeDataLoader(processingTypeConfigsLoader)
-      materializer             = Materializer(system)
-      _               <- Resource.eval(IO(JavaClassVersionChecker.check()))
-      _               <- Resource.eval(IO(SLF4JBridgeHandlerRegistrar.register()))
+
+      dbRef           <- DbRef.create(alreadyLoadedConfig.rawConfig)
       metricsRegistry <- createGeneralPurposeMetricsRegistry()
-      db              <- DbRef.create(alreadyLoadedConfig.rawConfig.resolved)
-      feStatisticsRepository <- QuestDbFEStatisticsRepository.create(
-        system,
-        clock,
-        alreadyLoadedConfig.rawConfig.resolved
+      dbioRunner = DBIOActionRunner(dbRef)(executionContextWithIORuntime)
+
+      infrastructureServices = InfrastructureServices(
```

Comment:
maybe we can move the "infrastructure services" creation to the factory method in the `InfrastructureServices` companion object? Like it's done in case of the "DomainServices" class